### PR TITLE
Lab 05. Finished main tasks

### PR DESCRIPTION
**Pull request by HenadzHatsko**
- Changed DAL to store metadata in database.
- Configured StoredFile entity to map it with database table.
- Implemented context initializer.

### CODE REVIEW

File: Lab05.EF/App.config [hunk 1]
```diff
@@ -0,0 +1,50 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <configSections>
+    <section name="log4net" type="log4net.Config.Log4NetConfigurationSectionHandler,log4net" />
+    <section name="entityFramework" type="System.Data.Entity.Internal.ConfigFile.EntityFrameworkSection, EntityFramework, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" requirePermission="false" />
+  </configSections>
+  <connectionStrings>
+    <add name="FileStorageConnection" connectionString="Data Source=(LocalDb)\MSSQLLocalDB;AttachDbFilename=C:\temp\Lab2_Proj\StorageDB.mdf;" providerName="System.Data.SqlClient" />
```
Comment (stas-litvinenka) [inline Lab05.EF/App.config]: Please put something like Lab05EF here. not Lab2_Proj

File: Lab05.EF/App.config [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      <layout type="log4net.Layout.PatternLayout">
+        <param name="ConversionPattern" value="%d  %-5p %m%n" />
+      </layout>
+    </appender>
+    <logger name="StorrageLogger">
+      <appender-ref ref="LogFileAppender" />
+    </logger>
+  </log4net>
+  <appSettings>
+    <add key="dbFile" value="C:\temp\Lab2_Proj\" />
```
Comment (stas-litvinenka) [inline Lab05.EF/App.config]: same here

File: Lab05.EF/Business/Services/Encryptor.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        {
+            try
+            {
+                encrypedPass = encrypedPass.Replace(" ", "+");
+                var cipherBytes = Convert.FromBase64String(encrypedPass);
+                using (var encryptor = Aes.Create())
+                {
+                    var pdb = new Rfc2898DeriveBytes(EncryptionKey,
+                        new byte[] {0x49, 0x76, 0x61, 0x6e, 0x20, 0x4d, 0x65, 0x64, 0x76, 0x65, 0x64, 0x65, 0x76});
+                    if (encryptor == null) return "\t Failed to encrypt.";
```
Comment (stas-litvinenka) [inline Lab05.EF/Business/Services/Encryptor.cs]: please use formating as in Encrypt method. Code is not structured here

File: Lab05.EF/Business/Services/SoredFileService.cs [hunk 4]
```diff
@@ -2,10 +2,10 @@
...
 using System.IO;
-using labFileStorage.Data.Entities;
-using labFileStorage.Data.Repositories;
+using Lab05.EF.Data.Entities;
+using Lab05.EF.Data.Repositories;
 
-namespace labFileStorage.Business.Services
+namespace Lab05.EF.Business.Services
 {
     public class SoredFileService : IFileService
```
Comment (stas-litvinenka) [inline Lab05.EF/Business/Services/SoredFileService.cs]: misspeling here. "t"

File: Lab05.EF/Business/Services/UserService.cs [hunk 5]
```diff
@@ -106,10 +106,7 @@ public string UserInfo()
 
                 foreach (var af in fileDictionary)
```
Comment (stas-litvinenka) [inline Lab05.EF/Business/Services/UserService.cs]: userStorrage = fileDictionary.Sum(x => x.Value.Size);  - looks better

File: Lab05.EF/Data/Repositories/Decorator.cs [hunk 6]
```diff
@@ -0,0 +1,53 @@
+﻿using System.Collections.Generic;
+
+namespace Lab05.EF.Data.Repositories
+{
+    public class Decorator<T> : BaseRepository<T> where T : class
+    {
+        protected BaseRepository<T> BaseRepository;
```
Comment (stas-litvinenka) [inline Lab05.EF/Data/Repositories/Decorator.cs]: lowercase here

File: Lab05.EF/Data/Repositories/Decorator.cs [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+        public Decorator(IPathHelper ph, BaseRepository<T> baseRepository) : base(ph)
+        {
+            BaseRepository = baseRepository;
```
Comment (stas-litvinenka) [inline Lab05.EF/Data/Repositories/Decorator.cs]: add check for null

File: Lab05.EF/App.config (hunk 1 above)
Comment (HenadzHatsko) [reply Lab05.EF/App.config]: Changed it.

File: Lab05.EF/App.config (hunk 2 above)
Comment (HenadzHatsko) [reply Lab05.EF/App.config]: Also changed.

File: Lab05.EF/Business/Services/Encryptor.cs (hunk 3 above)
Comment (HenadzHatsko) [reply Lab05.EF/Business/Services/Encryptor.cs]: Added a line to separate "if" block from other code.

File: Lab05.EF/Business/Services/SoredFileService.cs (hunk 4 above)
Comment (HenadzHatsko) [reply Lab05.EF/Business/Services/SoredFileService.cs]: Corrected

File: Lab05.EF/Business/Services/UserService.cs (hunk 5 above)
Comment (HenadzHatsko) [reply Lab05.EF/Business/Services/UserService.cs]: Used it.

File: Lab05.EF/Data/Repositories/Decorator.cs (hunk 7 above)
Comment (HenadzHatsko) [reply Lab05.EF/Data/Repositories/Decorator.cs]: Added one.

File: Lab05.EF/Data/Repositories/Decorator.cs (hunk 6 above)
Comment (HenadzHatsko) [reply Lab05.EF/Data/Repositories/Decorator.cs]: Changet to lowercase and made private.

File: Lab05.EF/Data/CurrentUser.cs [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+        //method compares entered username with the correct one
+        public bool UserNameCheck(string enteredName)
+        {
+            user = userRepository.Get(enteredName);
+            return user != null;
+        }
+
+        //method compares entered password with the correct one
+        public bool PasswordCheck(string enteredPassword)
```
Comment (stas-litvinenka) [inline Lab05.EF/Data/CurrentUser.cs]: move to separate place. entity is not the right place put logic in. Better : create class  e.g. AuthenticationService and appropriate method there- e.g. Authenticate(string password, encryptedpassword)

File: Lab05.EF/Data/CurrentUser.cs [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            return enteredPassword.Equals(EncryptionService.Decrypt(user.Password), StringComparison.CurrentCulture);
+        }
+
+        public string GetName()
```
Comment (stas-litvinenka) [inline Lab05.EF/Data/CurrentUser.cs]: remove. it is the same as using User.Name . no need to encapsulate this.

File: Lab05.EF/Data/CurrentUser.cs [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            return UserName;
+        }
+
+        public string GetDirectory()
```
Comment (stas-litvinenka) [inline Lab05.EF/Data/CurrentUser.cs]: remove. it is the same as using User.Directory. no need to encapsulate this.

File: Lab05.EF/Data/CurrentUser.cs [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            return UserDirectory;
+        }
+
+        public string GetInfo(long userStorrage)
```
Comment (stas-litvinenka) [inline Lab05.EF/Data/CurrentUser.cs]: create separate class for this

File: Lab05.EF/Data/CurrentUser.cs [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            return "\tlogin: " + UserName +
+                   "\n\tcreation date:" + CreationDate +
+                   $"\n\tstorage used: {userStorrage} bytes";
+        }
+
+        public void AuthenticationPassed(string enteredName)
```
Comment (stas-litvinenka) [inline Lab05.EF/Data/CurrentUser.cs]: move to separate class

File: Lab05.EF/Data/ICurrentUser.cs [hunk 13]
```diff
@@ -1,15 +1,17 @@
...
+﻿namespace Lab05.EF.Data
 {
-    public interface IUser
+    public interface ICurrentUser
     {
         string GetInfo(long userStorrage);
 
         string GetName();
 
+        string GetDirectory();
```
Comment (stas-litvinenka) [inline Lab05.EF/Data/ICurrentUser.cs]: remove

File: Lab05.EF/Data/ICurrentUser.cs [hunk 14]
```diff
@@ -1,15 +1,17 @@
-﻿namespace labFileStorage.Data
+﻿namespace Lab05.EF.Data
 {
-    public interface IUser
+    public interface ICurrentUser
     {
         string GetInfo(long userStorrage);
 
         string GetName();
```
Comment (stas-litvinenka) [inline Lab05.EF/Data/ICurrentUser.cs]: remove

File: Lab05.EF/Data/ICurrentUser.cs [hunk 15]
```diff
@@ -1,15 +1,17 @@
...
 
+        string GetDirectory();
+
         bool UserNameCheck(string enteredName);
 
         bool PasswordCheck(string enteredPassword);
 
-        void AuthenticationPassed(string enteredName, string enteredPassword);
+        void AuthenticationPassed(string enteredName);
```
Comment (stas-litvinenka) [inline Lab05.EF/Data/ICurrentUser.cs]: put somewhere else

File: Lab05.EF/Data/Repositories/EfUserRepository.cs [hunk 16]
```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using Lab05.EF.Data.Entities;
+
+namespace Lab05.EF.Data.Repositories
+{
+    public class EfUserRepository : BaseRepository<User>
+    {
+
```
Comment (stas-litvinenka) [inline Lab05.EF/Data/Repositories/EfUserRepository.cs]: remove empty line

File: Lab05.EF/Data/CurrentUser.cs (hunk 8 above)
Comment (HenadzHatsko) [reply Lab05.EF/Data/CurrentUser.cs]: moved to CurrentUserService and changed name

File: Lab05.EF/Data/CurrentUser.cs (hunk 9 above)
Comment (HenadzHatsko) [reply Lab05.EF/Data/CurrentUser.cs]: Fixed.

File: Lab05.EF/Data/CurrentUser.cs (hunk 10 above)
Comment (HenadzHatsko) [reply Lab05.EF/Data/CurrentUser.cs]: Also fixed.

File: Lab05.EF/Data/CurrentUser.cs (hunk 11 above)
Comment (HenadzHatsko) [reply Lab05.EF/Data/CurrentUser.cs]: Moved to CurrentUserService.

File: Lab05.EF/Data/CurrentUser.cs (hunk 12 above)
Comment (HenadzHatsko) [reply Lab05.EF/Data/CurrentUser.cs]: Moved to CurrentUserService and changed name to Authenticate

File: Lab05.EF/Data/ICurrentUser.cs (hunk 13 above)
Comment (HenadzHatsko) [reply Lab05.EF/Data/ICurrentUser.cs]: Removed interface.

File: Lab05.EF/Data/ICurrentUser.cs (hunk 14 above)
Comment (HenadzHatsko) [reply Lab05.EF/Data/ICurrentUser.cs]: Removed interface.

File: Lab05.EF/Data/ICurrentUser.cs (hunk 15 above)
Comment (HenadzHatsko) [reply Lab05.EF/Data/ICurrentUser.cs]: Removed interface.

**Pull request merged**